tests/leaderboards/test_tasks_leaderboard.py

```diff
@@ -0,0 +1,471 @@
+
+    This function compares each task's configuration from the `all_results` dictionary,
+    against the configuration which was used for evaluation.
+    Configuration is a flat dictionary, so the comparison is done at the lowest level.
+
+    Args:
+        evaluation_results (Tuple[ConfigParser, Dict]): A tuple containing the evaluation configuration
+        and the results for all tasks evaluated.
+
+    Examples of keys and values compared in this test:
+        - "model": "hf"
+        - "model_revision": "main"
+        - max_batch_size": 1
+    """
+    config, all_results = evaluation_results
+
+    # Parameters to exclude from comparison as they are not present in the evaluation output
+    # These parameters are only used to set up the evaluation
+    excluded_params = [
+        "evaluation_tracker_args",
+        "system_instruction",
+        "chat_template",
+        "max_batch_size",
+        "write_out",
+        "check_integrity",
+        "log_samples",
+        "apply_chat_template",
+        "fewshot_as_multiturn",
+        "verbosity",
+        "predict_only",
+    ]
+    # Prepare the expected configuration dictionary, it's task independent
+    expected_config = {
+        k: v for k, v in config.params.items() if k not in excluded_params
+    }
+    config_name = expected_config.pop("name")
+
+    for task_name in config.tasks.keys():
+        results = all_results[task_name]
+        compare_results(
+            reference=expected_config,
+            observed=results["config"],
+            config_name=config_name,
+            module_name="config",
+            recursive=False,
+        )
+
+
+def test_tasks_configs(evaluation_results: Tuple[ConfigParser, Dict]):
+    """
+    Compare the task configurations against the observed task configurations.
+
+    This function compares each task's detailed configuration from the `all_results` dictionary,
+    against the expected configuration for each task.
+    If the configuration has more than one key, i.e. task consists of multiple subtasks,
+    the function filters out only the subtasks of the currently considered task.
+
+    Args:
+        evaluation_results (Tuple[ConfigParser, Dict]): A tuple containing the evaluation configuration
+        and the results for all tasks evaluated.
+
+    Examples of configuration structure compared in this test:
+        - "fewshot_delimiter": "\n\n"
+        - "num_fewshot": 0
+        - "output_type": "generate_until"
+    """
+    config, all_results = evaluation_results
+
+    for task_name, task_configs in config.tasks.items():
+        results = all_results[task_name]
+        is_multitask = len(results["configs"]) > 1
+        subtasks_configs = (
+            filter_dict(task_configs.to_dict(), task_name)
+            if is_multitask
+            else {
+                task_name: {
+                    k: v for k, v in task_configs.to_dict().items() if k != "limit"
+                }
+            }
+        )
```

Comment:
Ternary expression is a bit long
```python
if is_multitask:
    substasks_config = {k: v for k, v in task_configs.to_dict.items() if task_name in k}
else:
    substasks_config = {
                task_name: {
                    k: v for k, v in task_configs.to_dict().items() if k != "limit"
                }
            }
```

Reply:
This is typically a function which could use more comments inside.

Why are you not removing limit in the first case?

Reply:
In multitask scenario, the limit is defined twice in the config: once as an argument for evaluation (which is removed to avoid mismatch), and once at a different level for output checking. In the first case, it can be used for both